### PR TITLE
CIV-6333, Defendant select immediately to pay

### PR DESCRIPTION
**Pull request by indu2019**
Email update sent to Claimant if defendant admits all and agrees to pay immediately

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-6333

### Change description ###
Email update sent to Claimant if defendant admits all and agrees to pay immediately

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantResponseApplicantNotificationHandler.java [hunk 1]
```diff
@@ -136,7 +137,13 @@ private void sendNotificationToSolicitor(CaseData caseData, String recipient) {
     private void sendNotificationToSolicitorSpec(CaseData caseData, String recipient, CaseEvent caseEvent) {
         String emailTemplate;
         if (caseEvent.equals(NOTIFY_APPLICANT_SOLICITOR1_FOR_DEFENDANT_RESPONSE)) {
-            emailTemplate = notificationsProperties.getClaimantSolicitorDefendantResponseForSpec();
+
+            if (caseData.getDefenceAdmitPartPaymentTimeRouteRequired().equals(IMMEDIATELY)) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantResponseApplicantNotificationHandler.java:141]:
```suggestion
            if (caseData.getDefenceAdmitPartPaymentTimeRouteRequired() == IMMEDIATELY) {
```

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantResponseApplicantNotificationHandler.java [hunk 2]
```diff
@@ -191,11 +198,24 @@ ALLOCATED_TRACK, toStringValueForEmail(caseData.getAllocatedTrack())
 
     public Map<String, String> addPropertiesSpec(CaseData caseData, CaseEvent caseEvent) {
         if (caseEvent.equals(NOTIFY_APPLICANT_SOLICITOR1_FOR_DEFENDANT_RESPONSE)) {
-            return Map.of(
+            if (caseData.getDefenceAdmitPartPaymentTimeRouteRequired().equals(IMMEDIATELY)) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/DefendantResponseApplicantNotificationHandler.java:201]:
```suggestion
            if (caseData.getDefenceAdmitPartPaymentTimeRouteRequired() == IMMEDIATELY) {
```

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java [hunk 3]
```diff
@@ -1681,8 +1682,15 @@ && ofNullable(caseData.getRespondent2Copy()).isPresent()) {
             updatedData.respondent2DetailsForClaimDetailsTab(updatedRespondent2);
         }
 
-        CaseRole respondentTwoCaseRoleToCheck;
+        if (caseData.getDefenceAdmitPartPaymentTimeRouteRequired() != null
+            &&  caseData.getDefenceAdmitPartPaymentTimeRouteRequired().equals(IMMEDIATELY)) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java:1686]:
```suggestion
            &&  caseData.getDefenceAdmitPartPaymentTimeRouteRequired() == IMMEDIATELY) {
```

Double Equal Sign (“==”) is null-safe whereas the equals() method is not. The equals() is a method of Enum class. If we try to call that method on a NULL reference we get a NullPointerException . 

“==” provides null-safety and type-safety during compile-time.

**Review by sabahirfan — DISMISSED** [review]

**Review by jeejavr — APPROVED** [review]

**Review by jeejavr — CHANGES_REQUESTED** [review]
shouldNotifyApplicantSolicitorSpecImmediately_whenInvoked() seems to be failing

**Pull request closed without merging**